src/odemis/driver/tfsbc.py

```diff
@@ -0,0 +1,357 @@
+# -*- coding: utf-8 -*-
+"""
+Created on 11 May 2020
+
+@author: Philip Winkler
+
+Copyright © 2020 Philip Winkler, Delmic
+
+This file is part of Odemis.
+
+Odemis is free software: you can redistribute it and/or modify it under the terms
+of the GNU General Public License version 2 as published by the Free Software
+Foundation.
+
+Odemis is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+Odemis. If not, see http://www.gnu.org/licenses/.
+"""
+from __future__ import division
+
+import logging
+
+import numpy
+import serial
+import serial.tools.list_ports
+from pymodbus.client.sync import ModbusSerialClient
+
+from odemis import model
+from odemis.model import HwError
+
+# Parameters for connection
+BAUDRATE = 230400
+TIMEOUT = 1
+BYTESIZE = 8
+PARITY = serial.PARITY_NONE
+STOPBITS = 1
+
+# Modbus level addresses
+SLAVE_UNIT = 2
+
+# Modbus registers
+BEAMDEFL_LX = 0  # lower x beam deflection element control
+BEAMDEFL_LY = 1  # lower y beam deflection element control
+BEAMDEFL_UX = 2  # upper x beam deflection element control
+BEAMDEFL_UY = 3  # upper y beam deflection element control
+
+# Transformation constants
+DCREVERSCOEF = -1
+DCROTUPPXX = 1
+DCROTUPPYY = 1
+DCROTLOWXX = -1
+DCROTLOWYY = 1
+
```

Comment:
Is there a reason you write it this way? Does it come from other code? All the values are always multiplied by DCREVERSCOEF, so at least it could be simplified by just multiplying them by -1 here.

Reply:
I took it from the other code. I also don't understand why they do it that way, but there must be a reason, so I just left it.

Reply:
If you copied it from the "recommended code", then it's fine to leave it as-is. Especially, if they update their code, it should be easy to update it here too.